lib/NetPacksLib.cpp

```diff
@@ -1419,9 +1418,8 @@ void GiveHero::applyGs(CGameState * gs) const
 		auto * boat = dynamic_cast<CGBoat *>(obj);
 		if (boat)
 		{
-			h->boat = boat;
-			h->attachTo(*boat);
-			boat->hero = h;
+			gs->map->removeBlockVisTiles(boat);

```

Comment:
Is this correct?
Here boat blockvis tiles are *removed* before attaching
But when we add boat on map init (CGameState.cpp change) its blockvis tiles are *added* before attaching

Reply:
If anything, the GameState is not correct as tiles will be double-blocked by hero and his boat.

Reply:
Tested with HoTA map which has a water Prison and it works fine as well.

Reply:
Looks like the question is if we should remove blockvis tiles of boat or of hero?

Reply:
Have you tested this hero disembarking and embarking on this boat later? And visiting this hero by another hero?

Reply:
I tried embarking, but not heroes visiting each other on boats, good idea.